app/build.gradle.kts

```diff
@@ -170,6 +170,9 @@ dependencies {
 
     // opening hours parser
     implementation("ch.poole:OpeningHoursParser:0.23.0")
+
+    // sunset-sunrise parser for lit quests
+    implementation("com.luckycatlabs:SunriseSunsetCalculator:1.2")
```

Comment:
I am always cautious when adding new dependencies. It looks like the calculation of that is not trivial, but did you review and compare all the different libraries that do implement that?

A short search revealed...
- https://github.com/shred/commons-suncalc
- https://github.com/caarmen/SunriseSunset
- https://github.com/mikereedell/sunrisesunsetlib-java (the one used by you)

The questions one gotta ask is whether they are well tested, whether they have any dependencies (they should not, if possible), whether it is maintained (an algorithm once written is finished, but maybe there are any bugs that are not addressed?) and finally, of course, whether it actually fulfills our needs.
(P.S: Java Date/Time classes from Java 8 **are** available in this app)

Reply:
https://github.com/caarmen/SunriseSunset seems to only include maven as a dependency, has some nice convenience methods and looks to be fairly well maintained, albeit not touched for a while. 

It's used by Telefónica, which gives me confidence in its abilities

https://github.com/shred/commons-suncalc has a few issues related to bugs, which have since been fixed but it doesn't give confidence in their testing. 

Feel free to criticise my judgement, I don't know a lot about Java so I'm not sure I'd be able to review the code for bugs myself.

TL;DR: Gonna switch to https://github.com/caarmen/SunriseSunset

Reply:
caarmen/SunriseSunset had a weird bug, had to not use it. Explain tomorrow :) It's currently working well tho

Reply:
Okay so caarmen's library war getting the wrong times for sunsets entirely. It was trying to tell me sunset was 2:02AM lat night for me when it was actually 20:45ish.

I have temporarily reverted back to the original library, but will take a look at the other one later today.

Reply:
Maybe the location entered was wrong? Switched lat and lon?

Reply:
Hey, finally something I have some expertise with!

Red Moon uses `com.luckycatlabs:SunriseSunsetCalculator`, mostly because it is the oldest and the only one available at the time when it was added. At some point, when the other libraries popped up, I did a pretty comprehensive review. Between that & my experience, my takeaways are:

- There is a fair amount of room for variation, because sunrise/sunset happen over a length of time.
- The different libs do not all use the same algorithm. Some are more accurate than others, but they are all approximations (Also, the amount of light hitting you varies depending on whether you're in a valley or mountain, etc, which these libs can't take into account due to technical constraints). Some of them perform better at different latitudes (closer/farther from the poles).
- `com.luckycatlabs:SunriseSunsetCalculator` is not the most accurate (I think — my memory is a little fuzzy). However, I decided that this didn't really matter for me. The inherent variation mentioned above was enough that even if it's off by 10 minutes from some ideal time to swap, it doesn't really matter.
    - The api has been extremely stable. I don't think I've had to modify the [calling code](https://github.com/LibreShift/red-moon/blob/master/app/src/main/java/com/jmstudios/redmoon/model/Config.kt#L173-L175)… ever? Maybe once. I think there were some bugfixes early on (none that affected me), but now it's pretty mature. I wouldn't have a problem with copying it into my code base if I wanted to avoid external dependencies entirely.\*

IIRC the [SunTimes](https://f-droid.org/en/packages/com.forrestguice.suntimeswidget/) app on F-Droid has an option to choose which library you want to use to calculate the time (currently it looks like just two options, luckycatlabs\*\* and caarmen, although I thought I remembered one or two others being included in the past), if you want to play around with it and see the differences.

\*Although I'd probably keep it in a separate library inside the same repo, like [how I have TimePickerPreference in Red Moon](https://github.com/LibreShift/red-moon/tree/master/timepickerpreference). Incidentally, @westnordost I wonder if it makes sense to use that kind of structure for your countryboundaries and osm… libraries (either moving them to this repo, or all to one repo for libraries; in any case, they can still be published as independent packages).

\*\*Technically it is a fork with a bugfix, but the bugfix has since been merged upstream and the dependency just wasn't switched back.

Reply:
Hearing good things about `com.luckycatlabs:SunriseSunsetCalculator` has made me feel better about my decision.

I have yet to test what happens if Sunrise or Sunset is before or after midnight, or what happens at the poles.